cmd/podman/search.go

```diff
@@ -153,8 +154,54 @@ func (s *searchParams) headerMap() map[string]string {
 	return values
 }
 
-func getSearchOutput(term string, registries []string, opts searchOpts, filter searchFilterParams) ([]searchParams, error) {
+// A wrapper for GetSystemContext and GetInsecureRegistries
+// Sets up system context and active list of registries to search with
+func getSystemContextAndRegistries(c *cli.Context) ([]string, *types.SystemContext, error) {
 	sc := common.GetSystemContext("", "", false)
+
+	// Variables for setting up Registry and TLSVerify
+	tlsVerify := c.BoolT("tls-verify")
+	forceSecure := false
+
+	if c.IsSet("tls-verify") {
+		forceSecure = c.BoolT("tls-verify")
+	}
+
+	var registries []string
+	if len(c.StringSlice("registry")) > 0 {
+		registries = c.StringSlice("registry")
+	} else {
+		var err error
+		registries, err = sysreg.GetRegistries()
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "error getting registries to search")
+		}
+	}
+
+	// If user flagged to skip verify for HTTP connections, set System Context as such
+	if !tlsVerify {
+		// If tls-verify is set to false, allow insecure always.
+		sc.DockerInsecureSkipTLSVerify = true
+	} else if !forceSecure {
+		// if the user didn't allow nor disallow insecure registries, check to see if the registry is insecure
+		insecureRegistries, err := sysreg.GetInsecureRegistries()
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "error getting insecure registries to search")
+		}
+
+		for _, reg := range insecureRegistries {
+			// if there are any insecure registries in registries, allow for HTTP
+			if util.StringInSlice(reg, registries) {
```

Comment:
no no No No NO *No* **NO**!

Just because _one_ registry is in the insecure list does not mean that it is OK to access any of the others insecurely as well.  This decision needs to be pushed all the way down to `docker.SearchRegistry` when we know which specific registry we are dealing with.

(Or, well, long-term, c/image/docker should do it automatically. Either way, this code needs to be replaced.)

Reply:
Oops. Should have caught this.
I guess we should be returning a separate SystemContext for each registry, individually setting InsecureSkipTLSVerify based on whether the registry is insecure, as well as the `tls-verify` flag.

Reply:
(Sorry about the drive-by comment, I don’t subscribe to all activity in all these projects. Maybe I should.)

Reply:
Interns, what are you going to do ¯\_(ツ)_/¯